src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs

```diff
@@ -2,30 +2,41 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using Microsoft.DotNet.Maestro.Client;
-using Microsoft.DotNet.Maestro.Client.Models;
-using Microsoft.Extensions.Logging;
-using Newtonsoft.Json.Linq;
-using NuGet.Packaging;
-using NuGet.Versioning;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Net;
-using System.Text;
+using System.Text.RegularExpressions;
 using System.Threading.Tasks;
 using System.Xml;
-using Microsoft.DotNet.Internal.Logging;
+using Microsoft.DotNet.Maestro.Client;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using NuGet.Versioning;
 
 namespace Microsoft.DotNet.DarcLib
 {
-    public class Remote : IRemote
+    public sealed class Remote : IRemote
     {
         private readonly IBarClient _barClient;
         private readonly GitFileManager _fileManager;
         private readonly IGitRepo _gitClient;
         private readonly ILogger _logger;
 
+        // **Updates**:
+        //- **Foo**: from to 1.1.0
+        //- **Bar**: from to 2.1.0
+        private static readonly Regex DependencyUpdatesPattern =
+            new Regex(@"- \*\*(?<updates>[\w+\.\-]+)\*\*: from (?<oldVersion>[\w\.\-]*) to (?<newVersion>[\w\.\-]+)");
+
+        // captures coherency updates from pr description, for eg: 
+        // **Coherency updates**:
+        //- **Foo**: from to 1.1.0
+        //- **Bar**: from to 2.1.0
+        private static readonly Regex CoherencyUpdatesPattern =
```

Comment:
This regex is literally identical to the previous one.. the only difference is the first one matches `- **pizza+**:  to b`, and the second does not.  That doesn't seem intentional, and then at that point why are their two expressions, especially since the first one matches ALL of the things the second does, and more, so much of the data is going to be duplicated.

I'm almost 100% certain these are not correct.

Also, the tests never test this minute difference.  It seems like a single expression (without all the matching groups) would work fine.

Reply:
redid the regex